nomad/structs/structs.go

```diff
@@ -1527,6 +1527,10 @@ func (s *Service) Validate() error {
 	}
 
 	for _, c := range s.Checks {
+		if s.PortLabel == "" && (c.Type == ServiceCheckTCP || c.Type == ServiceCheckHTTP) {
```

Comment:
@dadgar I was wondering the same, but I thought since service block holds check blocks, it's the service block which is invalid if the check can't be supported because the service doesn't have certain capability. Validation of a check currently checks whether they have been configured correctly, for example, if a user specifies the type of the check as `script`, then `command` is mandatory.